### PR TITLE
DRILL-5208: Finding path to java executable should be deterministic

### PR DESCRIPTION
**Pull request by paul-rogers**
See DRILL-5208 for background. Instead of using “find” to locate the
java command, we use the any information available, resorting to find
only if the “usual suspects” fails. The result is that we use the JDK
java when available, instead of randomly choosing JDK or JRE java.

### CODE REVIEW

File: distribution/src/resources/drill-config.sh [hunk 1]
```diff
@@ -367,12 +375,18 @@ if [ -z "$JAVA_HOME" ]; then
 fi
 
 # Now, verify that 'java' binary exists and is suitable for Drill.
-if $is_cygwin; then
-  JAVA_BIN="java.exe"
-else
-  JAVA_BIN="java"
+# If we started with `which java` above, use that path (after replacing
+# symlinks. If we started with JAVA_HOME, try in bin. Doing so handles
```
Comment (arina-ielchiieva) [inline distribution/src/resources/drill-config.sh]: Close the brace.

File: distribution/src/resources/drill-config.sh [hunk 2]
```diff
@@ -359,6 +366,7 @@ if [ -z "$JAVA_HOME" ]; then
       [[ $SOURCE != /* ]] && SOURCE="$DIR/$SOURCE"
     done
     JAVA_HOME="$( cd -P "$( dirname "$SOURCE" )" && cd .. && pwd )"
+    JAVA = $SOURCE/$JAVA_BIN
```
Comment (arina-ielchiieva) [inline distribution/src/resources/drill-config.sh]: $SOURCE already includes bin/java, so concatenating with $JAVA_BIN will give you:
`/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.111-1.b15.el7_2.x86_64/jre/bin/java/java`.
I guess you should concatenate $JAVA_BIN to $SOURCE dirname.

**Comment by paul-rogers** [comment]
Adressed code review comments and rebased onto latest master.

**Comment by sudheeshkatkam** [comment]
Please squash the commits as well.

**Comment by arina-ielchiieva** [comment]
+1, looks good.

**Pull request closed without merging**